src/main/java/org/alfresco/repo/management/JmxDumpUtil.java

```diff
@@ -164,9 +169,136 @@ private static void printMBeanInfo(MBeanServerConnection connection, ObjectName
+    {
+        //Selects the whole string, if one of the redactedInputs are present. It will have two capture groups, group 1 is the proceeding token ie. "password=" and group 2 will be all characters following the = (the vaule/password).
+        String regex = createPasswordFindRegexString(redactedInputs); 
+
+        //Replace the whole string with just capture group 1 to remove the desired value and concat the protected value.
+        String output = input.replaceAll(regex, "$1"+PROTECTED_VALUE);
+
+
+        return output;
+    }
+
+    /**
+     * Creates a regular expression that will select a string that contains one of the values provided in argEndings, proceeding an "=" and defines two capture groups:
+     * <ul>
+     * <li>Group 1: An argEnding that is followed by an "=", including the "=" and all character prior to the argEnding.
+     * <li>Group 2: The characters that follow group 1, to the end of the string or new line.
+     * </ul>
+     * <p>
+     * The argEnding can be the whole Input argument or the common characters proceeding the = sign.
+     * Example argEndings:
+     * <ul>
+     * <li>    -Ddb.password   This will select the values passed as -Ddb.password
+     * <li>    password        This will select any potential values that end in the word password
+     * </ul>
+     * <p>
+     * Example usage: 
+     * <p>
+     * argEndings={"password", "pwd"}
+     * <p>
+     * This will create a regex that will match a string that contains either argEndings. 
+     * The following will be matched by the resulting regex:
+     * <p>
+     * "-Ddb.password=my_password"
+     * <p>
+     * For this example: group 1="-Ddb.password=" group 2="my_password"
+     * 
+     * 
+     * @param argEndings Strings that will end the input argument you wish to select
+     * @return Regex pattern for selecting the characters following the strings passed as argEndings
+     */
+    static String createPasswordFindRegexString(String[] argEndings)
+    {
+        if(argEndings.length<1)
+        {
+            //throw Exception e = new Exception();
+        }
+
+        //(?i)(.*(arg[0]=|arg[1]=|...|arg[n-1]=))((?<=arg[0]=|arg[1]=|...|arg[n-1]=).*+)
+        String part1 = "(?i)";
+        String part2 = "(.*(";
+        String part3 = "((?<=";
+        
+        part2 += escapeRegexMetaChars(argEndings[0]) + "="; //(.*(arg[0]=
+        part3 += escapeRegexMetaChars(argEndings[0]) + "="; //((?<=arg[0]=
+
+        for (int i = 1; i < argEndings.length; i++) 
+        {
+            String arg = escapeRegexMetaChars(argEndings[i]);
+            part2 += "|" + arg + "="; //(.*(arg[0]=|arg[1]=arg[2]....|arg[n-1]=
+            part3 += "|" + arg + "="; //((?<=arg[0]=|arg[1]=|arg[2]...|arg[n-1]=
+        } 
+
+
+        part2 += "))"; //(.*(arg[0]=|arg[1]=|...|arg[n-1]=))
+        part3 += ").*+)"; //((?<=arg[0]=|arg[1]=|...|arg[n-1]=).*+)
+        String regex = part1 + part2 + part3;
+        return regex;

```

Comment:
Let's use something more elegant for string concatenation (maybe StringJoiner), make it easier to read/understand. 
I recommend this short article https://www.baeldung.com/java-strings-concatenation

Reply:
Creation of the string is now much simpler to read.